api/routes/participant.py

```diff
@@ -248,3 +248,39 @@ async def get_participants(
         external_participant_ids=external_participant_ids,
         internal_participant_ids=internal_participant_ids,
     )
+
+
+@router.post(
+    '/{participant_id}/update-participant-family',
+    operation_id='updateParticipantFamily',
+)
+async def update_participant_family(
+    participant_id: int,
+    old_family_id: int,
+    new_family_id: int,
+    connection: Connection = get_projectless_db_connection,
+):
+    """
+    Change a participants family from old_family_id
+    to new_family_id, maintaining all other fields.
+    The new_family_id must already exist.
+    """
+    player = ParticipantLayer(connection)
+
+    # Save current family_participant values to reinsert them
+    fp_row = await player.get_family_participant_data(
+        family_id=old_family_id, participant_id=participant_id
+    )
+
+    await player.remove_participant_from_family(
+        family_id=old_family_id, participant_id=participant_id
+    )
+
+    # Use saved values to maintain the fields in the new row
+    return await player.add_participant_to_family(
+        family_id=new_family_id,
+        participant_id=participant_id,
+        paternal_id=fp_row['paternal_id'],
+        maternal_id=fp_row['maternal_id'],
+        affected=fp_row['affected'],
+    )
```

Comment:
Can you move all this business logic to the layer. This allows it to be unit-tested.
It would also be good to wrap this around a database transaction to ensure a roll-back if an error occurs (like the new family ID doesn't exist).

Reply:
Added a new layer function specifically with this logic + a transaction wrapper, now the endpoint calls that layer function only